File: base/doc/guides/PrimaryProducerExample.cpp
#include "rgma/PrimaryProducer.h"
#include "rgma/RGMAException.h"
#include "rgma/Storage.h"
#include "rgma/SupportedQueries.h"
#include "rgma/TimeInterval.h"
#include "rgma/TimeUnit.h"

#include <string>
#include <iostream>

using glite::rgma::PrimaryProducer;
using glite::rgma::RGMAException;
using glite::rgma::Storage;
using glite::rgma::SupportedQueries;
using glite::rgma::TimeInterval;
using glite::rgma::TimeUnit;

int main(int argc, char* argv[])
{

   if (argc != 2)
   {
      std::cout<< "Usage: PrimaryProducerExample <userId>\n"<< std::endl;
      exit(1);
   }

   std::string args(argv[1]);

   try
   {
      PrimaryProducer pp (Storage::MEMORY, SupportedQueries::C);

      std::string table("default.userTable");
      std::string predicate("WHERE userId = '" + args + "'");
      TimeInterval historyRetentionPeriod(60, TimeUnit::MINUTES);
      TimeInterval latestRetentionPeriod(60, TimeUnit::MINUTES);
      pp.declareTable(table, predicate, historyRetentionPeriod,
            latestRetentionPeriod);

      std::string insert;
      insert.append("INSERT INTO default.userTable (userId, aString, ");
      insert.append("aReal, anInt) VALUES ('");
      insert.append(args);
      insert.append("', 'C++ producer', 3.1415926, 42)");
      pp.insert(insert);

      pp.close();
   }

   catch (RGMAException e)
   {
      std::cerr << "R-GMA exception: " << e.getMessage() << std::endl;
   }

}

File: base/doc/guides/PrimaryProducerExample.java
import org.glite.rgma.PrimaryProducer;
import org.glite.rgma.RGMAException;
import org.glite.rgma.Storage;
import org.glite.rgma.SupportedQueries;
import org.glite.rgma.TimeInterval;
import org.glite.rgma.TimeUnit;

public class PrimaryProducerExample {
  public static void main(String[] args) {

    if (args.length != 1) {
      System.err.println("Usage: java PrimaryProducerExample <userId>");
      System.exit(1);
    }

    String userId = args[0];

    try {
      PrimaryProducer producer = new PrimaryProducer(Storage
          .getMemoryStorage(), SupportedQueries.C);

      String predicate = "WHERE userId = '" + userId + "'";
      TimeInterval historyRetentionPeriod = new TimeInterval(60,
          TimeUnit.MINUTES);
      TimeInterval latestRetentionPeriod = new TimeInterval(60,
          TimeUnit.MINUTES);
      producer.declareTable("default.userTable", predicate,
          historyRetentionPeriod, latestRetentionPeriod);

      String insert = "INSERT INTO default.userTable "
          + "(userId, aString, aReal, anInt)" + " VALUES ('" + userId
          + "', 'Java producer', 3.1415926, 42)";
      producer.insert(insert);

      producer.close();

    } catch (RGMAException e) {
      System.err.println("R-GMA exception: " + e.toString());
      System.exit(1);
    }

  }
}
